oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/ElasticRequestHandler.java

```diff
@@ -390,32 +392,26 @@ private MoreLikeThisQueryBuilder mltQuery(Map<String, String> mltParams) {
             mlt = moreLikeThisQuery(fieldsArray, null, new Item[]{new Item(null, text)});
         }
 
-        for (String key : mltParams.keySet()) {
+        BiConsumer<String, Consumer<String>> mltParamSetter = (key, setter) -> {
             String val = mltParams.get(key);
-            if (MoreLikeThisHelperUtil.MLT_MIN_DOC_FREQ.equals(key)) {
-                mlt.minDocFreq(Integer.parseInt(val));
-            } else if (MoreLikeThisHelperUtil.MLT_MIN_TERM_FREQ.equals(key)) {
-                mlt.minTermFreq(Integer.parseInt(val));
-            } else if (MoreLikeThisHelperUtil.MLT_BOOST_FACTOR.equals(key)) {
-                mlt.boost(Float.parseFloat(val));
-            } else if (MoreLikeThisHelperUtil.MLT_MAX_DOC_FREQ.equals(key)) {
-                mlt.maxDocFreq(Integer.parseInt(val));
-            } else if (MoreLikeThisHelperUtil.MLT_MAX_QUERY_TERMS.equals(key)) {
-                mlt.maxQueryTerms(Integer.parseInt(val));
-            } else if (MoreLikeThisHelperUtil.MLT_MAX_WORD_LENGTH.equals(key)) {
-                mlt.maxWordLength(Integer.parseInt(val));
-            } else if (MoreLikeThisHelperUtil.MLT_MIN_WORD_LENGTH.equals(key)) {
-                mlt.minWordLength(Integer.parseInt(val));
-            } else if (MoreLikeThisHelperUtil.MLT_MIN_SHOULD_MATCH.equals(key)) {
-                mlt.minimumShouldMatch(val);
-            } else if (MoreLikeThisHelperUtil.MLT_STOP_WORDS.equals(key)) {
-                // TODO : Read this from a stopwords text file, configured via index defn maybe ?
-                String[] stopWords = val.split(",");
-                mlt.stopWords(stopWords);
-            } else {
-                LOG.warn("Unrecognized param {} in the mlt query {}", key, mltParams);
```

Comment:
The LOG.warn is now lost...

What about using switch / case instead of BiConsumer?

Reply:
True. I did remove it because I realized it was logging stuff, for every query, that should not be logged. For example, `stream.body` and `mlt.fl` (fields list). I have also switched from `for` to `BiConsumer` to avoid iterating/checking the same fields when a single lookup would be OK.
I have revisited the code removing the params from the map. If, in the end, there are still entries left the WARNING is printed. To avoid modifying the original map I have created a shallow one (more in the comment).